File: backend/gustsim/resources/read_results.m
% MATLAB: h5read works directly; no OpenFOAM installation is required.
filename = 'fields.h5';
manifest = jsondecode(h5readatt(filename, '/', 'manifest_json'));
points = h5read(filename, '/mesh/points')';  % MATLAB reverses HDF5 dimensions
connectivity = double(h5read(filename, '/mesh/connectivity')) + 1;
offsets = double(h5read(filename, '/mesh/offsets'));
pressure = h5read(filename, '/fields/cell/pressure_pa');
velocity = h5read(filename, '/fields/cell/U')';
firstCell = connectivity(offsets(1)+1:offsets(2));
speed = vecnorm(velocity, 2, 2);
fprintf('Run %s: %d points, pressure [%g, %g] Pa gauge\n', ...
    manifest.run_id, size(points,1), min(pressure), max(pressure));

File: examples/read_results.py
"""python read_results.py fields.h5 -- no OpenFOAM installation required."""
import json
import sys
import h5py
import numpy as np

with h5py.File(sys.argv[1]) as results:
    manifest = json.loads(results.attrs['manifest_json'])
    xyz = results['mesh/points'][:]
    cells = results['mesh/connectivity'][:]
    offsets = results['mesh/offsets'][:]
    pressure = results['fields/cell/pressure_pa'][:]
    velocity = results['fields/cell/U'][:]
    print('Run:', manifest['run_id'])
    print('Points:', xyz.shape, 'Pressure range [Pa gauge]:', pressure.min(), pressure.max())
    print('First cell vertex indices:', cells[offsets[0]:offsets[1]])
    print('Speed range [m/s]:', np.linalg.norm(velocity, axis=1).min(), np.linalg.norm(velocity, axis=1).max())
